EventFilter/TotemRawToDigi/plugins/TotemVFATRawToDigi.cc

```diff
@@ -0,0 +1,160 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors: 
+*   Jan Kašpar (jan.kaspar@gmail.com)
+*
+****************************************************************************/
+
+#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "DataFormats/TotemDigi/interface/TotemRPDigi.h"
+#include "DataFormats/TotemDigi/interface/TotemVFATStatus.h"
+#include "DataFormats/TotemDigi/interface/TotemFEDInfo.h"
+
+#include "CondFormats/DataRecord/interface/TotemReadoutRcd.h"
+#include "CondFormats/TotemReadoutObjects/interface/TotemDAQMapping.h"
+#include "CondFormats/TotemReadoutObjects/interface/TotemAnalysisMask.h"
+
+#include "EventFilter/TotemRawToDigi/interface/SimpleVFATFrameCollection.h"
+#include "EventFilter/TotemRawToDigi/interface/RawDataUnpacker.h"
+#include "EventFilter/TotemRawToDigi/interface/RawToDigiConverter.h"
+
+#include <string>
+
+//----------------------------------------------------------------------------------------------------
+
+class TotemVFATRawToDigi : public edm::one::EDProducer<>
```

Comment:
this can be made edm::stream
We should not loose CPU efficiency in data processing without a good reason. ::one will make all threads to wait until it's done.